ios-sdk/core-data.md

```diff
@@ -0,0 +1,180 @@
+
+### Add data using Core Data Framework
+
+In your view where you are saving the data to the table. lets import the core data framework so that we can use all the related methods in the given viewController file.
+
+```js
+import CoreData
+```
+
+Now lets create a function that will save the data to the database.
+
+```js
+func save(name: String, star: Int16) {
+
+  // Step 1 create app delegate
+  guard let appDelegate = UIApplication.shared.delegate as? AppDelegate
+  else {
+    return
+  }
+
+  // Step 2 create an instance of NSManagedObjectContext
+  let managedContext = appDelegate.persistentContainer.viewContext
+
+  // Step 3 create an instance of NSManagedObjectContext which
+  // just fetches all the attribute information for the given entity
+  // using NSEntityDescription and then creates a variable called
+  // repo which contains all the data required.
+
+  let entity = NSEntityDescription.entity(forEntityName: "Repo",
+                                          in: managedContext)!
+
+  let repo = NSManagedObject(entity: entity,
+                             insertInto: managedContext)
+  print("\(repo), REPO")
+
+  // Step 2 will return a Repo with name and star set to the default,
+  // (for me it will set it to nil for the name and 0 for the star,
+  // it could be different if you've set your default value to something different)
+  // We change this to the new ones that are being passed into this
+  // function in Step 3
+  task.setValue(name, forKeyPath: "name")
+  task.setValue(star, forKeyPath: "star")
+
+  // Step 4 we save the data using the .save() method. Please note
+  // that is recommended that you put this in a try catch block so that you
+  // don't crash the app if you run into an error when saving the data.
+  do {
+      try managedContext.save()
+  } catch let error as NSError {
+      print("Could not save. \(error), \(error.userInfo)")
+  }
+
+}
+```
+There you have it, if you run this function in the place where you are saving the data, it will now store this data in our database!
+
+### Remove data using Core Data Framework
+
+Now that we have the ability to save data, we can add another function that will allow us to delete the data.
+
+```js
+
+guard let appDelegate = UIApplication.shared.delegate as? AppDelegate
+else {
+    return
+}
+
+let managedContext = appDelegate.persistentContainer.viewContext
+
+repo.remove(at: indexPath.row)
+
+```
+
+You'll notice that our first couple of steps are very similar to when we are saving the data, Once we create an appDelegate and create a variable that store the context all we need to do is call `remove(at:Int)` where you enter the row number that you'd like to delete. For me as I am using a Table view I would write the above function as `remove(at:indexPath.row)`
```

Comment:
Weird capitalisation after this comma.

It might be useful to split your paragraphs out across various lines in README's in future so that it is easier to refer/link to a specific sentence in that paragraph as well.